server/chat-commands/core.ts

```diff
@@ -1324,28 +1357,30 @@ exports.commands = {
 				}));
 				return false;
 			}
-			let roomList = {};
+			let roomList: any = {};
+			// has to be disabled because roomid is reassigned, but only in an if statement.
+			// eslint-disable-next-line prefer-const
 			for (let roomid of targetUser.inRooms) {
 				if (roomid === 'global') continue;
-				let targetRoom = Rooms.get(roomid);
+				const targetRoom = Rooms.get(roomid);
 				if (!targetRoom) continue; // shouldn't happen
-				let roomData = {};
+				const roomData: AnyObject = {};
 				if (targetRoom.isPrivate) {
 					if (!user.inRooms.has(roomid) && !user.games.has(roomid)) continue;
 					roomData.isPrivate = true;
 				}
 				if (targetRoom.battle) {
-					let battle = targetRoom.battle;
+					const battle = targetRoom.battle;
 					roomData.p1 = battle.p1 ? ' ' + battle.p1.name : '';
 					roomData.p2 = battle.p2 ? ' ' + battle.p2.name : '';
 				}
 				if (targetRoom.auth && targetUser.id in targetRoom.auth) {
-					roomid = targetRoom.auth[targetUser.id] + roomid;
+					(roomid as string) = targetRoom.auth[targetUser.id] + roomid;
 				}
 				roomList[roomid] = roomData;
 			}
 			if (!targetUser.connected) roomList = false;
-			let userdetails = {
+			const userdetails: Record<string, any> = {
```

Comment:
```suggestion
			const userdetails: AnyObject = {
```

I don't know who taught you `Record`, but PS writes this as `{[k: string]: any}`. Or, we would, but we already have a type alias for that: `AnyObject`.

Reply:
oh, strange, npm test was complaining to use that